Create issues page on about page

lanieerts opened the pull request.

lanieerts committed:
Seperate reducer about comment and modify query.

lanieerts committed:
Modify detail of query for readMoreComments

lanieerts committed:
Modify url of action for createComment.

lanieerts committed:
Merge remote-tracking branch 'upstream/master'

pjhjohn committed:
Rebase commit of upstream/master.
Introduce react-router-bootstrap to avoid page refresh

Removed legacy style for Tab. Removed navbar radius by overriding style

Resolved #32 : Implemented Sticky Footer

Reference : https://getbootstrap.com/examples/sticky-footer

Disabled default visibility of ReduxDevTools

CRLF -> LF

reference : http://stackoverflow.com/questions/1889559/git-diff-to-ignore-m

Comment, PostDetail, Post re-styled with react-bootstrap

* Enabled well class in _bootstrap.scss

UI Enhancement using Panel Component

lanieerts committed:
Merge remote-tracking branch 'upstream/master'

lanieerts committed:
Merge remote-tracking branch 'upstream/master'

lanieerts committed:
Merge branch 'master' of https://github.com/lanieerts/jsonplaceholder-client

lanieerts committed:
Merge remote-tracking branch 'upstream/master'

lanieerts committed:
Merge remote-tracking branch 'upstream/master'

lanieerts committed:
Add issues page on about page.

pjhjohn commented on src/components/GithubIssues/GithubIssues.js:
```diff
@@ -0,0 +1,18 @@
+
+import { Panel } from 'react-bootstrap';
+
+class GithubIssues extends React.Component {
```
하나의 깃헙 이슈네요. 단수형을 써주세요!

pjhjohn commented on src/components/GithubIssues/GithubIssues.js:
```diff
@@ -0,0 +1,18 @@
+  render() {
+    return (
+      <div>
+        <Panel header={"Number : " + this.props.number} bsStyle="primary">
```
[ES6 Template Literals](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Template_literals)를 씁시다!

pjhjohn commented on src/components/GithubIssues/GithubIssues.js:
```diff
@@ -0,0 +1,18 @@
+    return (
+      <div>
+        <Panel header={"Number : " + this.props.number} bsStyle="primary">
+          Title : {this.props.title} <br />
```
`<br />` 은 지양합시다.

pjhjohn commented on src/components/index.js:
```diff
@@ -6,3 +6,4 @@ export Comment from './Comment/Comment';
 export PhotoDetail from './Photos/PhotoDetail';
 export PhotoItem from './Photos/PhotoItem';
 export AlbumItem from './Albums/AlbumItem';
+export GithubIssues from './GithubIssues/GithubIssues';
```
`GitHub`에 관련된 컴포넌트들은 하나로 둘 수 있을테니, 아래와 같은 구조는 어떨까요??

```
export GithubIssue from './Github/GithubIssue';
```

pjhjohn commented on src/components/index.scss:
```diff
@@ -6,3 +6,4 @@
 @import './Photos/PhotoDetail';
 @import './Photos/PhotoItem';
 @import './Albums/AlbumItem';
+@import './GithubIssues/GithubIssues';
```
위와 동일

pjhjohn commented on src/containers/About/About.js:
```diff
@@ -1,52 +1,20 @@
-        <Panel header="Issues" bsStyle="warning">
-          <Inspector data={this.props.issues} />
-        </Panel>
+        <Link to="about/issues">
```
보통 `/`으로 시작합니다

pjhjohn commented on src/containers/About/Issues/Issues.js:
```diff
@@ -0,0 +1,72 @@
+import { GithubIssues } from './../../../components';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import { readIssues } from './../../../actions';
```
`react`&`redux` 라이브러리와 관계 없는 우리 프로젝트의 import는 **한 줄 띄우고** 넣습니다

pjhjohn commented on src/containers/About/Issues/Issues.js:
```diff
@@ -0,0 +1,72 @@
+import { bindActionCreators } from 'redux';
+import { readIssues } from './../../../actions';
+
+import { DropdownButton, MenuItem, ListGroup, ListGroupItem } from 'react-bootstrap';
```
`node_modules`에 있는 모듈을 import하는 것이므로 위의 `./../../../actions` 보다 위로 가야합니다.

pjhjohn commented on src/containers/About/Issues/Issues.js:
```diff
@@ -0,0 +1,72 @@
+
+import { DropdownButton, MenuItem, ListGroup, ListGroupItem } from 'react-bootstrap';
+
+class Issues extends React.Component {
```
`About`의 하위기는 하지만 아직까지 굳이 디렉토리를 하나 더 들어갈 필요는 없을 것 같습니다.
`src/containers/About/Issues/Issues.js`
=> `src/containers/About/Issues.js` 혹은 `src/containers/About/GithubIssues.js` 는 어떤가요?

pjhjohn commented on src/containers/About/About.js:
```diff
@@ -1,52 +1,20 @@
 class About extends React.Component {
 
-  componentWillMount() {
-    this.props.readLanguages();
```
임시로 없애더라도 없앨 때는 쓰이지 않는 코드가 생기지 않도록 관련된 코드들을 모두 제거해주세요!
or
어딘가에 쓰이도록 원래 있던 부분들은 어느정도 살려두거나

lanieerts replied:
action의 readLanguage, readContributors는 나중에 추가될 때 사용 될 것 같아 그대로 두었습니다.

lanieerts committed:
Modify issues about #48.
1. Rename GithubIssues component.
2. Using ES6 templete literals.
3. Replace tag to <p> from <br>.
4. Move directory about GithubIusse component and Issues container.
5. Add / in link path.
6. Clean up import order.

pjhjohn commented on src/containers/About/Issues.js:
```diff
@@ -0,0 +1,73 @@
+          <h2> State : {this.state.state} / Sort : {this.state.sort} </h2>
+        </div>
+        <ListGroup>
+          {
```
```
{
  this.props.issues.map((issues) =>
```

to
`{ this.props.issues.map((issues) =>`

pjhjohn commented on src/containers/About/Issues.js:
```diff
@@ -0,0 +1,73 @@
+    'sort': 'created'
+  };
+
+  updateState = (state) => {
```
이거 `updateState` 랑 `updateSort` 를 따로 분리해서 value만 받는 것 보다
`updateFilter` 처럼 만들어서 `axios`에서 `query`로 받는 것 처럼 하는게 버튼이 늘어날 수록 좋을듯.
어차피 `this.setState`는 업데이트용 함수이기 때문에 오히려 코드가 간결해질 것임.

lanieerts replied:
버튼이 늘어나면 늘어날수록 onclick 이벤트의 update함수에 들어갈 parameter가 점점 더 길어져서 더 복잡해질 것 같습니다만....
그리고 setState를 안 쓰면 다른 옵션에 대한 값을 불러올 방법이 생각이 안 나네요. 좋은 방법 있나요?

pjhjohn commented on src/containers/About/Issues.js:
```diff
@@ -0,0 +1,73 @@
+    this.setState({
+      'sort': sort
+    });
+    this.props.readIssues(Object.assign({ 'state': this.state.state }, { 'sort': sort }));
```
여기서도 위의 변화가 긍정적인 결과를 가져오는데,
`this.state.state` 보다 `this.state.filter` 나 `this.state.issueFilter` 같은게 더 가독성이 좋을 것 같슴다

pjhjohn commented on src/containers/About/Issues.js:
```diff
@@ -0,0 +1,73 @@
+            <MenuItem onClick={() => this.updateSort('updated')}> UPDATED </MenuItem>
+            <MenuItem onClick={() => this.updateSort('comments')}> COMMENTS </MenuItem>
+          </DropdownButton>
+          <h2> State : {this.state.state} / Sort : {this.state.sort} </h2>
```
Use ES6 Template Literals

pjhjohn commented on src/components/Github/GithubIssue/GithubIssue.js:
```diff
@@ -0,0 +1,18 @@
+
+import { Panel } from 'react-bootstrap';
+
+class GithubIssues extends React.Component {
```
클래스명도 단수화 시켜줘야 :D
그리고, `Github` 아래로 옮기자는 뜻은 `Github/GithubIssue/GithubIssue` 가 아니라 `Github/GithubIssue` 였습니다!

pjhjohn commented on src/components/Github/GithubIssue/GithubIssue.js:
```diff
@@ -0,0 +1,18 @@
+    return (
+      <div>
+        <Panel header={`Number : ${this.props.number}`} bsStyle="primary">
+          <p> Title : {this.props.title} </p>
```
이 부분들도 되도록이면 `ES6 Template Literals`를 씁시당!

lanieerts replied:
여길 어떻게 바꿔야되나요?

pjhjohn replied:
```
<p>{`Title : ${this.props.title}`}</p>
```

pjhjohn commented on src/components/Github/GithubIssue/GithubIssue.js:
```diff
@@ -0,0 +1,18 @@
+class GithubIssues extends React.Component {
+  render() {
+    return (
+      <div>
```
불필요한 `<div>` 입니다.

pjhjohn commented on src/containers/About/About.js:
```diff
@@ -1,52 +1,19 @@
 class About extends React.Component {
-
-  componentWillMount() {
-    this.props.readLanguages();
```
더이상 사용하지 않는 `action`&`reducer` 는 함께 삭제해주세용

lanieerts committed:
Modify additional issues about #48.
1. Combine filter function.
2. Apply ES6 templete literals.
3. Rename class name of GithubIssues -> GithubIssue
4. Delete unused div tag.
5. Delete unused actions & reducers.

pjhjohn commented on src/components/Github/GithubIssue.js:
```diff
@@ -0,0 +1,16 @@
+  render() {
+    return (
+      <Panel header={`Number : ${this.props.number}`} bsStyle="primary">
+        <p> {`Title : ${this.props.title}`} </p>
```
`<Tag>` 의 앞뒤는 스페이스를 자제하는게 좋습니당 나중에 프로젝트 전체 일괄적으로 확인할게요~

pjhjohn commented on src/components/index.scss:
```diff
@@ -6,3 +6,4 @@
 @import './Photos/PhotoDetail';
 @import './Photos/PhotoItem';
 @import './Albums/AlbumItem';
+@import 'Github/GithubIssue';
```
`./`으로 시작하기

lanieerts replied:
으엌 이건 왜...

pjhjohn commented on src/containers/About/Issues.js:
```diff
@@ -0,0 +1,71 @@
+        <h2> ISSUES PAGE </h2>
+        <div>
+          <DropdownButton title="STATE" bsStyle="primary" id="state">
+            <MenuItem onClick={() => this.updateFilter({ state: 'open' })}> DEFAULT </MenuItem>
```
DEFAULT는 왜 있는건가염?

lanieerts replied:
쓸 때는 필요 할 거라고 생각했는데 지금 생각하니 아니네요(...)
LABEL의 경우, 아무것도 없을 때가 검색 조건에 포함될 것 같아서 남겨 놓겠습니다.

pjhjohn commented on src/containers/About/Issues.js:
```diff
@@ -0,0 +1,71 @@
+class Issues extends React.Component {
+
+  state = {
+    filter: {}
```
첨에 filter가 없을 때는 버튼 중 무엇과도 매치가 되지 않는 것인가요?

lanieerts replied:
네. state.filter에 아무것도 없어요. 그 때도 ComponentWillMount에서 불러 올 수는 있습니다.
